### ISSUE TITLE
Installing new sdk versions through dotnet

### ISSUE THREAD

**Issue by lewislabs** [dotnet/cli#5371]
I'm wondering if this is achievable, it's certainly something that's desirable from a portability standpoint. 

If I specify an `"sdk":{ "version": "x.y.z"}` in the `global.json`, and i run `dotnet build` on a machine without the `x.y.z` version of the sdk, could `dotnet` go and install that version?

The main benefit of this is developer productivity and consistency of the build tools used across development environments for any given project.

**Comment by piotrpMSFT** [comment]
This is certainly something we want to enable, though we won't be able to get to it for 1.0.0.

**Comment by lewislabs** [comment]
👍 thanks for the information @piotrpMSFT

**Comment by RheaAyase** [comment]
I do not think that this is technically possible _everywhere_ (remember the multi-platform aspect?) Linux permissions.

**Comment by tmds** [comment]
> I do not think that this is technically possible everywhere (remember the multi-platform aspect?) Linux permissions.

@RheaAyase Windows has permissions too :smile:

**Comment by RheaAyase** [comment]
@tmds Well, at least we can see how much I know about Windows. :D

**Comment by livarcocc** [comment]
We have been talking about adding a `dotnet install sdk` command that could be used explicitly by the user to acquire an SDK. The first turn of the crank we have been talking about making this part of the SDK itself, which means the host would not be able to do that for you.

**Comment by RheaAyase** [comment]
No installing of anything via dotnet please. Bad bad bad thing.

**Comment by svick** [comment]
@RheaAyase Do you think it would be acceptable if `dotnet install sdk` just delegated to the existing platform installer?

So, on Ubuntu, `dotnet install sdk 2.0.0` would run something like `apt-get install dotnet-sdk-2.0.0` (and would need to be run with `sudo`). On RHEL, it would run the appropriate `yum` command. On Windows, it would download and run the installer.

**Comment by RheaAyase** [comment]
> Do you think it would be acceptable if dotnet install sdk just delegated to the existing platform installer?

No it is not, I mean... KISS right? Why complicate the lives of dotnet engineers, what if someone will decide to create a new package manager tomorrow that won't follow the usual conventions? As you mention permissions are also a thing and running dotnet with `sudo` is a big no-no itself. It's overall really bad idea: it's a lot of unnecessary work, many potential issues, and it brings no benefit to the user at all - none...
